components/solicitation/Step.tsx

```diff
@@ -0,0 +1,76 @@
+import type { Status } from "./FollowSolicitation.tsx";
+import Icon from "$store/components/ui/Icon.tsx";
+
+export interface Props extends Status {
+  statusIndex: number;
+  index: number;
+  isLastStep: boolean;
+}
+
+function Step({ title, statusIndex, toolTip, index, isLastStep }: Props) {
+  const isStepFinish = index < statusIndex;
+  const isCurrentStep = index === statusIndex;
+  const isStepForward = index > statusIndex;
+  const isFirst = index === 0;
+
+  return (
+    <li
+      class={`max-lg:flex-grow flex-shrink-[1] lg:w-48 ${
+        (isFirst || isLastStep) ? "max-md:w-min" : ""
+      }`}
```

Comment:
Voce pode facilitar lógicas usando funcionalidades do CSS:
```suggestion
      class={`max-lg:flex-grow flex-shrink-[1] lg:w-48 first:max-md:w-min last:max-md:w-min`}
```